hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -185,7 +185,9 @@ private void balance() {
     for (int i = 0; i < idleIteration && balancerRunning; i++) {
       // stop balancing if iteration is not initialized
       if (!initializeIteration()) {
-        stop();
+        if (isBalancerRunning()) {
```

Comment:
i think there may be a very corner case.
```
    if (isBalancerRunning()) {
// at this time , `stop` may be called from command line and the lock is hold.
// so, the `stop` here will stuck in a deadlock
      stop();
    }
```

Reply:
I was thinking about this too. Can we do anything to protect against this situation?